### PR TITLE
 Peephole optimizer for "iszero iszero <tag> jumpi". 

### PR DESCRIPTION
**Pull request by chriseth**
Fixes https://github.com/ethereum/solidity/issues/6212

### CODE REVIEW

File: test/cmdlineTests/gas_test_dispatch_optimize/output [hunk 1]
```diff
@@ -2,52 +2,52 @@
 ======= gas_test_dispatch_optimize/input.sol:Large =======
 Gas estimation:
 construction:
-   300 + 259200 = 259500
+   300 + 260400 = 260700
```
Comment (axic) [inline test/cmdlineTests/gas_test_dispatch_optimize/output:5]: Why did this one increase? :)

Comment (chriseth) [reply test/cmdlineTests/gas_test_dispatch_optimize/output:5]: This is weird. I analyzed it a little. The thing is that the peephole optimizer is the first thing that is run. This means in the two versions, the CSE step will have a different input. The newly generated code contains many more literal zeros and before this change, there were more DUPi to retrieve the zero from further down the stack.

File: test/cmdlineTests/gas_test_dispatch_optimize/output [hunk 2]
```diff
@@ -2,52 +2,52 @@
...
+   g6(uint256):	20620
+   g7(uint256):	20730
+   g8(uint256):	20708
+   g9(uint256):	20554
 
 ======= gas_test_dispatch_optimize/input.sol:Medium =======
 Gas estimation:
 construction:
-   183 + 140200 = 140383
+   183 + 140800 = 140983
```
Comment (leonardoalt) [inline test/cmdlineTests/gas_test_dispatch_optimize/output:33]: This one also increased.

**Review by leonardoalt — APPROVED** [review]

**Review by leonardoalt — CHANGES_REQUESTED** [review]
Needs rebasing.
Also one test says it's failing, but when I open it I don't see the error.

**Comment by chriseth** [comment]
Test is not failing anymore.

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**